jaeger_client/senders.py

```diff
@@ -28,14 +31,66 @@
 
 
 class Sender(object):
-    def __init__(self, host, port, io_loop=None):
-        self.host = host
-        self.port = port
+    def __init__(self, io_loop=None):
+        from threading import Lock
         self.io_loop = io_loop or self._create_new_thread_loop()
+        self._process_lock = Lock()
+        self._process = None
+        self.spans = []
+
+    def append(self, span):
+        """Queue a span for subsequent submission calls to flush()"""
+        self.spans.append(span)
+
+    @property
+    def span_count(self):
+        return len(self.spans)
+
+    @tornado.gen.coroutine
+    def flush(self):
+        """
+        Examine span and process state before yielding to _flush() for batching and transport.
+        Returns a tuple with any raised Exception and error message info yielded from _flush().
+        """
+        exc = None
+        error_msg = ''
+        if self.spans:
+            with self._process_lock:
+                process = self._process
+            if process:
+                exc, error_msg = yield self._flush(self.spans, self._process)
+                self.spans = []
+
+        raise tornado.gen.Return((exc, error_msg))
+
+    @tornado.gen.coroutine
+    def _flush(self, spans, process):
+        """
+        Batch spans and invokes send() while providing all exception handling.
+        Override with specific batching logic and error messaging, if desired.
+        """
+        exc = None
+        error_msg = ''
+        batch = thrift.make_jaeger_batch(spans=spans, process=process)
+
+        try:
+            yield self.send(batch)
+        except Exception as e:
+            exc = e
```

Comment:
Here you can just `raise e("my message") from e` and it's done! (alongwith the previous comment modifications)

The `from e` part preserve the exception stacktrace

Reply:
`from e` is not supported in python 2.

Reply:
Yeah, you're right. I'm too used to python3